soc/xtensa/intel_adsp/common/cavs_ipc.c

```diff
@@ -49,6 +49,7 @@ void z_cavs_ipc_isr(const void *devarg)
 
 		regs->tdr = CAVS_IPC_BUSY;
 		if (done && !IS_ENABLED(CONFIG_SOC_SERIES_INTEL_CAVS_V15)) {
+			k_busy_wait(100);
```

Comment:
I'm afraid this is way, way too long sleep. We have to run a gazillion tasks during every 1ms and IPCs can be sent at any time, so the IPC ISR simply needs to be fast.

Reply:
Looking at the differences to https://github.com/thesofproject/sof/blob/main/src/drivers/intel/cavs/ipc.c , at least masking of the BUSY irq is missing here (while the message is handled). Not likely to explain this case as the message is handled directly in the ISR, but at least one difference.

Reply:
It's possible that this is some secondary behavior related to the test that SOF doesn't do too.  Maybe it's turning around and spinning on the registers in a way that SOF doesn't, and putting the delay into the test might be sufficient?

The masking seems needless to me: this is the ISR, so the interrupt won't (can't) reentrantly fire until we return (absent the handler playing games with the interrupt level or other outrageous hacks).  And once the ISR returns, we know it's not latched anymore because we don't enter an infinite interrupt cycle.  Even if the hardware does fire one during the command contra the spec, it should look different in a way (a spurious new command) that this test should detect and fail on.  Certainly timing won't be able to fix that.

But masking is cheap, so no reason not to.